client/src/components/_common/TextCopyField/TextCopyField.module.scss

```diff
@@ -1,5 +1,14 @@
 .root {
   display: flex;
+  .copy-button {
+    transition: background-color 5s;
+  }
+}
+
+.root.copied {
```

Comment:
The reason this does not work __might__ be the parent class change (unlikely), or __maybe__ the CSS modules middleware (sad). Can you try performing this with class change on the button?

P.S. Sample code (but you have the right direction, already): https://codepen.io/tacc-wbomar/pen/MWjGqaG

Reply:
Yeah pretty sure it's CSS modules. It doesn't compute the class name correctly. I tried doing it directly on the button but it doesn't seem to grab a ref on it properly. I may have to do something nasty like another component inside of that one to get a proper ref.

Reply:
Even though the issue is __not__ CSS transition via parent class (I tested on [follow-up pen](https://codepen.io/tacc-wbomar/pen/PoGedXJ)), perhaps simplifying the translation CSS Modules needs to do would work (no need for another component).

```
.root {
  // whatever, unrelated
}

.copy-button {
  transition: background-color 5s;
}

.copy-button.is-copied {
  background-color: green;
}
```

P.S. A component just for the transition seems like it would simplify the CSS in the same way.